AK/StdLibExtras.h

```diff
@@ -43,38 +44,6 @@ void compiletime_fail(Args...);
 #    define AK_REPLACED_STD_NAMESPACE std
 #endif
 
-namespace AK_REPLACED_STD_NAMESPACE { // NOLINT(cert-dcl58-cpp) Names in std to aid tools
-
-// NOTE: These are in the "std" namespace since some compilers and static analyzers rely on it.
-//       If USING_AK_GLOBALLY is false, we can't put them in ::std, so we put them in AK::replaced_std instead
-//       The user code should not notice anything unless it explicitly asks for std::stuff, so...don't.
-
-template<typename T>
-constexpr T&& forward(AK::Detail::RemoveReference<T>& param)
```

Comment:
This commit both moves stuff around and adds new things. When moving code, it's almost always better to make one commit that does the move and nothing else, and another that changes / adds things.

Unrelatedly, let's not include `<memory>` all over the place. Can't we implement `construct_at` the same way libc++ / libstdc++ do?

Reply:
> Unrelatedly, let's not include <memory> all over the place. Can't we implement construct_at the same way libc++ / libstdc++ do?

This will have to be guarded with `AK_DONT_REPLACE_STD` and the flag itself is very problematic (technically, it changes ABI, so if someone ever wants to use our code with std, they need to recompile everything with `AK_DONT_REPLACE_STD`).

Reply:
Luckily we don't care about anyone but us using our code :)

Reply:
If only we didn't mix AK and std ourselves (in ports)

Reply:
>Unrelatedly, let's not include <memory> all over the place. Can't we implement construct_at the same way libc++ / libstdc++ do?

I try to only include the internal header, but I have no clue how that one is called in the wasm stdlib, and so on,
and the standard tells us to include `<memory>` for that and the function we need is sadly "magic" and must be in the std namespace to work

Reply:
found and added the libcxx header, which should lessen the impact

Reply:
> This commit both moves stuff around and adds new things.

It still does this. Please make one commit for move and another for adding things, as requested above.

Reply:
> for that and the function we need is sadly "magic" and must be in the std namespace to work

I'm willing to believe that this could be true, but could you explain in some more detail what would break if we added

```
template <typename T, class... Args, class = decltype(::new(declval<void*>()) T(declval<Args>()...))> constexpr T* construct_at(T* address, Args&&... args) {
  VERIFY(__location);
  return ::new (static_cast<void*>(address)) T(forward<Args>(args)...);
}
```

instead?

Reply:
> would break if we added

This:
> if someone ever wants to use our code with std, they need to recompile everything with AK_DONT_REPLACE_STD

You will have compilation error with your suggestion + `#include <memory>` in the same translation unit.

Reply:
I think thakis means the extra constraints and the Verify, which I can add no problem

Reply:
p.s. there's no guarantee that the compiler will treat the placement new as constexpr (gcc trunk does in c++26, gcc 14.2 does not): https://gcc.godbolt.org/z/6Wvr3c1G7

Reply:
Well according to <https://en.cppreference.com/w/cpp/compiler_support>
GCC only added support in version 15
(also there are two FTMs)